### ISSUE TITLE
Scene loader for 2D scenes

### ISSUE THREAD

**Issue by Alicemargatroid** [mosra/magnum#171]
We have COLLADA, OBJ, OpenGEX, and PLY for scene loading, and I saw that ASSIMP and other loaders are being considered, but what about for 2D scenes?

I've seen a lot of 2D games using [Tiled](http://www.mapeditor.org/), which exports its own XML format called [TMX](http://doc.mapeditor.org/reference/tmx-map-format/), as well as a few other exports (the Lua one might conform to our OpenDDL parser). 

If 2D is supposed to not be an "ugly stepchild", why does its older brother 3D get all the nice things?

**Comment by mosra** [comment]
Good point you have there :)

The thing is that I didn't know ~~much~~ actually, any formats for 2D scenes -- just 2D images. I briefly considered SVG as a 2D scene format, but I don't really see it mapping well to the use case. 

TMX is new to me. Seems to be pretty simple. I expected some COLLADA-like mess when I saw XML, but no, it's not! :)

There might be also a few things missing from the actual data classes for 2D stuff in `Trade` namespace -- I don't have much experience with 2D scenes so it feels okay for me, but maybe you already have ideas what could be improved, what's missing etc.

**Comment by renindustries** [comment]
@mosra Nah, it is straight forward XML, very easy to parse. Because it is designed to be parsed from the top down (there is no need to do more than one pass), a SAX parser would work.

I'll take a look at `Trade`, I'll investigate to see if there is a header-file only SAX XML parser that can be integrated, and get back to you (maybe with a PR if all goes well).

**Comment by Alicemargatroid** [comment]
@mosra Nah, it is straight forward XML, very easy to parse. Because it is designed to be parsed from the top down (there is no need to do more than one pass), a SAX parser would work.

I'll take a look at `Trade`, I'll investigate to see if there is a header-file only SAX XML parser that can be integrated, and get back to you (maybe with a PR if all goes well).

**Comment by mosra** [comment]
Oh my, awesome :) Thanks!

No need to go all header-only, I'm completely okay with depending on some well-known third-party library if it is easily portable (or already ported) to Android/iOS/Emscripten. I think libxml2 would fit this description, though it is just plain C with no fancy C++11 features. There is even `FindLibXml2.cmake` as part of the vanilla CMake installation and there is an Emscripten version at https://github.com/kripken/xml.js/, so I think it might be the least painful way.

I used Qt4 for the COLLADA importer, but I don't want to touch that framework ever again ;)

**Comment by feliwir** [comment]
I recommend pugixml, it's fast and portable

**Comment by mosra** [comment]
Tiled importer is being done in mosra/magnum-plugins#59.
